### ISSUE TITLE
no-unused-var doesn't trigger when using rest operator in assignement 

### ISSUE THREAD

**Issue by johangirod** [babel/babel-eslint#248]
### How to reproduce
Using `babel-eslint@5.0.0-beta10`
**.eslintrc**
```json
{
    "parser": "babel-eslint",
    "env": {
        "es6": true,
    },
    "parserOptions": {
        "ecmaFeatures": {
            "experimentalObjectRestSpread": true
        },
    },
    "rules": {
        "no-unused-vars": 2,
    }
}
```
**test.js**
```js
const { a, ...b } = { a: 1, b: 2, c: 3 };
```

When executing
```shell
$ eslint test.js
```
The linter does not raise any errors. It should have printed : 
```
test.js
  1:9   error  'a' is defined but never used  no-unused-vars
  1:15  error  'b' is defined but never used  no-unused-vars
```
(which it does with the eslint parser)

**Comment by hzoo** [comment]
Check https://github.com/babel/babel-eslint/issues/95 - it was changed to be like this.

**Comment by johangirod** [comment]
Thanks for the explanation !
I guess you can close this.

**Comment by hzoo** [comment]
Ok cool closing
